src/main/java/com/sprint/mission/discodeit/controller/UserController.java

```diff
@@ -0,0 +1,130 @@
+package com.sprint.mission.discodeit.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sprint.mission.discodeit.dto.binaryContent.BinaryContentCreateRequest;
+import com.sprint.mission.discodeit.dto.user.UserCreateRequest;
+import com.sprint.mission.discodeit.dto.user.UserResponse;
+import com.sprint.mission.discodeit.dto.user.UserUpdateRequest;
+import com.sprint.mission.discodeit.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestPart;
+
+
+import java.io.IOException;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+    private final UserService userService;
+
+
+    @RequestMapping(value = "/create", method = RequestMethod.POST, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<UserResponse> createUser(
+            @RequestPart("userCreateRequest") String userCreateRequestStr,
+            @RequestPart(value = "profile", required = false) MultipartFile profile) {
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            UserCreateRequest userCreateRequest = objectMapper.readValue(userCreateRequestStr, UserCreateRequest.class);
+
+            BinaryContentCreateRequest imageRequest = null;
+            if (profile != null && !profile.isEmpty()) {
+                imageRequest = new BinaryContentCreateRequest(
+                        profile.getOriginalFilename(),
+                        profile.getContentType(),
+                        profile.getBytes()
+                );
+            }
+
+            UserResponse createdUser = userService.createUserWithProfileImage(userCreateRequest, imageRequest);
+            return ResponseEntity.status(HttpStatus.CREATED).body(createdUser);
+
+        } catch (IOException e) {
+            throw new IllegalArgumentException("프로필 이미지 처리 중 오류 발생: " + e.getMessage());
+        }
+    }
+
+    @RequestMapping(value = "", method = RequestMethod.GET)
+    public ResponseEntity<UserResponse> getUserById(@RequestParam("userid") UUID userId) {
```

Comment:
userid는 RequestParam 보단 PathVariable로 받는게 좀 더 자연스러울 것 같네요.  
그리고 userid는 스네이크패턴 유지해서 userId로 수정해야할 것 같아요.  
```suggestion
    public ResponseEntity<UserResponse> getUserById(@PathVariable("userId") UUID userId) {
```